File: java/AbstraktniTrida/src/main/java/org/example/Programator.java
package org.example;

public class Programator extends Clovek {

    //private String jazyk;
    private JazykTyp jazyk;

    public Programator(String jmeno, int vek, JazykTyp jazyk) {

        super(jmeno, vek);
        this.jazyk = jazyk;
    }

    @Override
    public String vratPovolani() {
        return "Programator";
    }

    @Override
    public void Pozdrav()
    {
        System.out.println("Ahoj posila " + jmeno + " ve veku " +  vek + " je dospely " + jeDospely + " ze tridy Programator programuje v " + jazyk);
    }

    @Override
    public String toString()
    {
        return jmeno + " " + vek + " " + jazyk;
    }

    public JazykTyp getJazyk() {
        return jazyk;
    }

    public void setJazyk(JazykTyp jazyk) {
        this.jazyk = jazyk;
    }
}

File: csharp/AbstraktniTrida/AbstraktniTrida/Programator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AbstraktniTrida
{
    internal class Programator: Clovek
    {
        //public string Jazyk { get; set; }
       public JazykTyp Jazyk { get; set;  }

        public Programator(string jmeno, int vek, JazykTyp jazyk) : base(jmeno, vek)
        {

            Jazyk = jazyk;
        }

        public override void Pozdrav()
        {
            Console.WriteLine($"Ahoj posila {Jmeno} ve veku {Vek} je dospely {jeDospely} ze tridy Programator programuje v {Jazyk}");
        }

        public override string ToString()
        {
            return $"{Jmeno} + {Vek} + {Jazyk}";
        }

        public override string VratPovolani()
        {
            return "Programator";
        }
    }
}
